homeassistant/helpers/config_entry_oauth2_flow.py

```diff
@@ -233,6 +242,80 @@ async def _token_request(self, data: dict) -> dict:
         return cast(dict, await resp.json())
 
 
+class LocalOAuth2ImplementationWithPkce(LocalOAuth2Implementation):
+    """Local OAuth2 implementation with PKCE."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        domain: str,
+        client_id: str,
+        authorize_url: str,
+        token_url: str,
+        client_secret: str = "",
+        code_verifier_length: int = 100,
```

Comment:
We need to validate the length to be between 43 and 128.

https://www.rfc-editor.org/rfc/rfc7636#section-4.1

Maybe default to 128?